File: Models/ViewModels/DriverBookingVM.cs
namespace GoWheels_WebAPI.Models.ViewModels
{
    public class DriverBookingVM : BaseModelVM
    {
        public DateTime PickUpDate { get; set; }
        public DateTime DropOffDate { get; set; }
        public decimal Total { get; set; }
        public bool IsCancel { get; set; }
        public DriverVM Driver { get; set; } = null!;

        //public List<InvoiceVM> Invoices { get; set; } = new List<InvoiceVM>();
    }
}

File: FE/src/Model/DriverBookingVM.js
import DriverVM from "./DriverVM";
import InvoiceVM from "./InvoiceVM"
export default class DriverBookingVM {
    constructor(data = {}) {
      
      this.pickUpDate = data.pickUpDate || "";
      this.dropOffDate = data.dropOffDate || "";
      this.total = data.total || 0;
      this.isCancel = data.isCancel || "";
      this.driverId = data.driverId || "";
      this.driver = (data.driver || []).map(driver => new DriverVM(driver));
      this.invoice = (data.invoice || []).map(invoice => new InvoiceVM(invoice));
    }
  }
  

